From: Hyeonggon Yoo <42.hyeyoo@gmail.com>
Subject: mm/zsmalloc: reset zpdesc fields in reset_zpdesc()
Date: Tue, 14 Jan 2025 00:29:52 +0900

To prepare for the future removal of struct page fields (e.g.  the removal
of page->index [1]), update reset_zpdesc() to reset the descriptor via
struct zpdesc instead of struct page.

As struct zpdesc overlays struct page for now, no functional change is
intended.

[1] https://lore.kernel.org/linux-mm/Z09hOy-UY9KC8WMb@casper.infradead.org

Link: https://lkml.kernel.org/r/Z4Uw136VdG7vlKCL@localhost.localdomain
Signed-off-by: Hyeonggon Yoo <42.hyeyoo@gmail.com>
Cc: Alex Shi <alexs@kernel.org>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Sergey Senozhatsky <senozhatsky@chromium.org>
Cc: Vishal Moola (Oracle) <vishal.moola@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/zsmalloc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/zsmalloc.c~mm-zsmalloc-convert-reset_page-to-reset_zpdesc-fix
+++ a/mm/zsmalloc.c
@@ -849,8 +849,8 @@ static void reset_zpdesc(struct zpdesc *
 
 	__ClearPageMovable(page);
 	ClearPagePrivate(page);
-	set_page_private(page, 0);
-	page->index = 0;
+	zpdesc->zspage = NULL;
+	zpdesc->next = NULL;
 	__ClearPageZsmalloc(page);
 }
 
_
